hazelcast/src/main/java/com/hazelcast/nio/serialization/FieldType.java

```diff
@@ -55,4 +71,32 @@ public static FieldType get(byte type) {
         return ALL[type];
     }
 
+    public boolean isArrayType() {
+        return type >= PORTABLE_ARRAY.type;
```

Comment:
Sure, I'm fully aware of it and I was expecting these comments :) 
It's kind of a common-sense solution since there's no more types in the portable supported. 
So it's not like this that there will be new types added.
I also added an explicit test case for it, so if somebody changes sth here the test will break too.
The advantage of the `isArrayType` and `getSingleType` is that they are blazing fast.
One is just an int comparison, the other one is a modulo. This code is invoked for each MapEntry so I was trying to keep it fast.
I disagree that it's easy to break it in this case (tests + no more types in portable) but I'm open to change if sb proposes a solution that's that fast and doesn't require an allocation of a Map.